sdk/trace/evictedqueue.go

```diff
@@ -19,13 +18,19 @@ type evictedQueue[T any] struct {
 	logDropped   func()
 }
 
-func newEvictedQueue[T any](capacity int) evictedQueue[T] {
-	var tVal T
-	msg := fmt.Sprintf("limit reached: dropping trace %T", tVal)
```

Comment:
The majority of this change is due to these lines. Local testing, removing these lines and leaving it as generic, results in about -300ns, -304B, and -4 Allocs. 

We (the sig) should check if this pattern is used elsewhere, as it's probably causing performance issues.